plugins/plugin-git/che-plugin-git-ext-git/src/main/java/org/eclipse/che/ide/ext/git/client/commit/CommitViewImpl.java

```diff
@@ -227,32 +217,112 @@ public void onMessageChanged(KeyUpEvent event) {
         delegate.onValueChanged();
     }
 
-    @UiHandler("addAll")
-    public void onAddAllValueChange(final ValueChangeEvent<Boolean> event) {
+    @UiHandler("amend")
+    public void onAmendValueChange(final ValueChangeEvent<Boolean> event) {
         if (event.getValue()) {
-            this.addSelection.setValue(false);
+            this.delegate.setAmendCommitMessage();
+        } else {
+            this.message.setValue("");
         }
+        delegate.onValueChanged();
     }
 
-    @UiHandler("addSelection")
-    public void onAddSelectionValueChange(final ValueChangeEvent<Boolean> event) {
-        if (event.getValue()) {
-            this.addAll.setValue(false);
-        }
+    @Override
+    public void setChangedPanelView(ChangedPanelView changedPanelView) {
+        this.render = new ChangedPanelRender(changedPanelView);
```

Comment:
The idea is to have an abbility to set custom render to `ChangesPanelView` to not to overload this widget with logic that is used only for commit window. Changes list window uses this panel with default render.